File: src/main/java/com/ptah/entity/accounting/Transaction.java
package com.ptah.entity.accounting;

import com.ptah.common.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.SourceType;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Transaction extends BaseEntity {

    @ManyToOne
    private Account account;
    @Enumerated(EnumType.STRING)
    private TransactionDirection direction;
    private BigDecimal amount;

}

File: src/main/kotlin/com/ptah/entity/accounting/Transaction.kt
package com.ptah.entity.accounting

import com.ptah.common.BaseEntity
import jakarta.persistence.*
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.math.BigDecimal

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
class Transaction : BaseEntity() {
    @ManyToOne
    private val account: Account? = null

    @Enumerated(EnumType.STRING)
    private val direction: TransactionDirection? = null
    private val amount: BigDecimal? = null
}
